### PR TITLE
Adding test cases for serving pkg

### LINKED ISSUE #1322: Increase code coverage to be larger than 80%
Codecov is now active and measures the code coverage which is currently ~ 73%

In order to meet the 1.0 criteria, we need to raise the coverage to 80%

The coverage with a drill down can be found at https://codecov.io/gh/knative/client/tree/main

### PR DESCRIPTION
**Pull request by vyasgun**
## Description
Adding test cases for serving pkg

## Changes

* Added new test cases
* Modified existing test cases for more coverage

## Reference

Related to #1322

### CODE REVIEW

File: pkg/serving/v1/client_test.go [hunk 1]
```diff
@@ -193,18 +209,90 @@ func TestUpdateService(t *testing.T) {
...
+		assert.ErrorContains(t, err, "conflict")
+	})
+
+	t.Run("updating a service with update func error", func(t *testing.T) {
+		_, err := client.UpdateServiceWithRetry(context.Background(), "update-service", func(svc *servingv1.Service) (*servingv1.Service, error) {
+			svc.Name = "update-service"
+			return svc, fmt.Errorf("update error")
+		}, 10)
+		assert.ErrorContains(t, err, "update error")
+	})
```
[the remark this answers is not in the document]
Comment (vyasgun) [reply pkg/serving/v1/client_test.go:253]: yeah, I could do that

[the remark this answers is not in the document]
Comment (dsimansk) [reply pkg/serving/v1/client_test.go:253]: Imo it's following the rest of test's style, but feel free to update it. :)

**Comment by dsimansk** [comment]
Thanks!
/approve
/lgmt

@itsmurugappan I'm going to approve this one, hope that's ok. If you feel like we should refactor more tests to table test cases maybe we can discuss that on WG call.

File: pkg/serving/v1/client_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (vyasgun) [reply pkg/serving/v1/client_test.go:253]: just took a look and a lot of tests will need to be refactored in the file for consistency. Maybe we can create another issue for it and address it later?

**Pull request merged**